pkg/chunk/grpc/grpc_server_mock.go

```diff
@@ -0,0 +1,203 @@
+		return &empty.Empty{}, nil
+	}
+	err := errors.New("delete table from tableClient request doesn't match with test gRPC client")
+	return &empty.Empty{}, err
+}
+
+func (s server) DescribeTable(ctx context.Context, name *TableName) (*DescribeTableResponse, error) {
+	if name.TableName == "chunk_2591" {
+		return &DescribeTableResponse{
+			Desc: &TableDesc{
+				Name:              "chunk_2591",
+				UseOnDemandIOMode: false,
+				ProvisionedRead:   0,
+				ProvisionedWrite:  0,
+				Tags:              nil,
+			},
+			IsActive: true,
+		}, nil
+	}
+	err := errors.New("describe table from tableClient request doesn't match with test gRPC client")
+	return &DescribeTableResponse{}, err
+}
+
+func (s server) UpdateTable(ctx context.Context, request *UpdateTableRequest) (*empty.Empty, error) {
+	if request.Current.Name == "chunk_2591" && !request.Current.UseOnDemandIOMode && request.Current.ProvisionedWrite == 0 &&
+		request.Current.ProvisionedRead == 0 && request.Current.Tags == nil && request.Expected.Name == "chunk_2591" &&
+		!request.Expected.UseOnDemandIOMode && request.Expected.ProvisionedWrite == 1 &&
+		request.Expected.ProvisionedRead == 300 && request.Expected.Tags == nil {
+		return &empty.Empty{}, nil
+	}
+	err := errors.New("update table from tableClient request doesn't match with test gRPC client")
+	return &empty.Empty{}, err
+}
+
+func (s server) Stop(ctx context.Context, empty *empty.Empty) (*empty.Empty, error) {
+	return empty, nil
+}
+
+// NewStorageClient returns a new StorageClient.
+func NewTestStorageClient(cfg Config, schemaCfg chunk.SchemaConfig) (*StorageClient, error) {
+	grpcClient, _, err := connectToTestGrpcServer(cfg.Address)
+	if err != nil {
+		return nil, err
+	}
+	client := &StorageClient{
+		schemaCfg: schemaCfg,
+		client:    grpcClient,
+	}
+	return client, nil
+}
+
+//***********************  gRPC mock server *********************************//
+
+// NewTableClient returns a new TableClient.
+func NewTestTableClient(cfg Config) (*TableClient, error) {
+	grpcClient, _, err := connectToTestGrpcServer(cfg.Address)
+	if err != nil {
+		return nil, err
+	}
+	client := &TableClient{
+		client: grpcClient,
+	}
+	return client, nil
+}
+
+// NewStorageClient returns a new StorageClient.
+func newTestStorageServer(cfg Config) (*server, error) {
+	client := &server{
+		Cfg: cfg,
+	}
+	return client, nil
+}
+
+func connectToTestGrpcServer(serverAddress string) (GrpcStoreClient, *grpc.ClientConn, error) {
+	params := keepalive.ClientParameters{
+		Time:                time.Second * 20,
+		Timeout:             time.Minute * 10,
+		PermitWithoutStream: true,
+	}
+	param := grpc.WithKeepaliveParams(params)
+	cc, err := grpc.Dial(serverAddress, param, grpc.WithInsecure())
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to dial grpc-store %s", serverAddress)
+	}
+	return NewGrpcStoreClient(cc), cc, nil
+}
+
+func createTestGrpcServer() {
+	var cfg server
+	lis, err := net.Listen("tcp", "localhost:6688")
```

Comment:
Don't hardcode ports. It makes tests more flaky. You can listen on `localhost:0`, and then having `createTestGrpcServer()` returning the address where the server is listening to.